language/stdlib/modules/libra_account.move

```diff
@@ -573,7 +577,18 @@ module LibraAccount {
         //     configuration such as AccountLimits before calling this
         move_to(&new_account, Role<RoleData> { role_data });
         // (3) publish Balance resource(s)
-        move_to(&new_account, Balance<Token>{ coin: Libra::zero<Token>() });
+        add_currency<Token>(&new_account);
+        if (add_all_currencies) {
+            if (!::exists<Balance<Coin1::T>>(new_account_addr)) {
+                add_currency<Coin1::T>(&new_account);
+            };
+            if (!::exists<Balance<Coin2::T>>(new_account_addr)) {
+                add_currency<Coin2::T>(&new_account);
+            };
+            if (!::exists<Balance<LBR::T>>(new_account_addr)) {
+                add_currency<LBR::T>(&new_account);
+            };
+        };
```

Comment:
Overall this looks god, and I'm fine with this going in as-is. 

One higher-level question I had though was whether we want this logic in module code, or to instead have this be in the transaction script code (i.e. `create_account(...)` and then have this logic within the script). What do you think?

Reply:
I think having it in the script could be good, but I don't see how to do it. The problem is that we need a `signer` to call `add_currency` on, but the only `signer` available in the script is the tx sender (not the `signer of the created account, which we have here). Do you have a scheme for this?

Reply:
ah! I didn't see this. It's doing a `move_to` into the newly created account (still wrapping my head around this `&signer` stuff...). This all makes a lot more sense now.

I can't think of any other way around doing this unless we allow `create_signer` to be a public function, which isn't a good idea.